FormidableLabs/react-native-app-auth
# feat: Add a config plugin for Expo (iOS)

Fixes N/A

## Description

This patch adds a new `plugin` directory with an Expo [config plugin](https://docs.expo.dev/config-plugins/introduction/).

This plugin will automatically patches the iOS Objective-C code to perform the different steps described in the README:
- it adds protocol conformance to `RNAppAuthAuthorizationFlowManager` to the `AppDelegate`
- it creates a new `authorizationFlowManagerDelegate` property
- it patches `application:openURL:options:` to handle the custom-scheme based redirects
- it patches `application:continueUserActivity:restorationHandler:` to handle the deep (universal) link redirects

⚠️ to patch the `AppDelegate.h` it makes use of the `withDangerousMod` and the "codeMod" API, which are not stable (but everybody uses them and we don't really have a choice here...)
 
## Steps to verify

1. create a new Expo project (any template will do)
2. add `react-native-app-auth` to the project: you can point to my branch for testing purposes:
    ```json
    "dependencies": {
        "react-native-app-auth": "aurelien-iapp/react-native-app-auth#expo-config-plugin",
        ...
    ```
4. add the plugin to your `app.json`'s `"plugins"` section:
    ```json
    "plugins": [
        "react-native-app-auth/plugin"
        ...
    ```
5. Generate the iOS/Android projects using `npx expo prebuild` and check the contents of `AppDelegate.mm` and `AppDelegate.h`. Changes will be visible as follows:

![AppDelegate h@2x](https://github.com/FormidableLabs/react-native-app-auth/assets/30048195/15bef56e-07c0-4413-bc15-5fe09c7a6a40)
![20231019 XcodeLeJeuduTao — AppDelegate mm@2x](https://github.com/FormidableLabs/react-native-app-auth/assets/30048195/c75fd266-59ac-4366-af37-3b80ab2b079f)
